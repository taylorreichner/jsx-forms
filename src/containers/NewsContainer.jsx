import React, { Component } from 'react';
import NewsArticleList from '../components/news/NewsArticlesList';
import { fetchNews } from '../services/newsApi';

export default class NewsContainer extends Component {
  state = {
    loading: true,
    articles: [],
  };

  async componentDidMount() {
    const articles = await fetchNews();
    this.setState({
      articles,
      loading: false,
    });
  }

  render() {
    const { loading, articles } = this.state;

    if (loading) return <h1>Loading...</h1>;

    return <NewsArticleList articles={articles} />;
  }
}
