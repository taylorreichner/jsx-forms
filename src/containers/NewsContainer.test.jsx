import React from 'react';
import { getAllByText, render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import NewsContainer from './NewsContainer';

jest.mock('../services/newsApi', () => ({
    fetchNews: () => [{ title: 'trees', text: 'tall trees', link: 'taylor'}],
    fetchNewsBySearch: () => [{title: 'trees', text: 'tall trees', link: 'taylor'}]
}))

describe('news Container', () => {
  it('displays a list of news', async () => {
    render(<NewsContainer />);

    screen.getByText('Loading...');

	const submitButton = screen.getByRole('button', { name: 'find-articles' });

	const searchForm = screen.getByRole('form', { name: 'form-input' });

	const articleList = await screen.findByRole('list', { name: 'articles' });

	expect(articleList).toMatchSnapshot();
	expect(submitButton).toMatchSnapshot();
	expect(searchForm).toMatchSnapshot();


	return waitFor(async () => {
		const searchList = await screen.findByRole('list', { name: 'articles' });

		expect(searchList.childElementCount).toEqual(1);
	});
  });
});