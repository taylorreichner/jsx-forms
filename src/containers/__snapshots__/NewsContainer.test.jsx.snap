// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`news Container displays a list of news 1`] = `
<ul
  aria-label="articles"
>
  <li>
    <figure>
      <figcaption>
        trees
      </figcaption>
      <figcaption>
        tall trees
      </figcaption>
      <figcaption>
        taylor
      </figcaption>
    </figure>
  </li>
</ul>
`;

exports[`news Container displays a list of news 2`] = `
<button
  aria-label="find-articles"
>
  Submit
</button>
`;

exports[`news Container displays a list of news 3`] = `
<form
  aria-label="form-input"
>
  <label
    for="article-search"
  >
    Search Articles
  </label>
  <input
    aria-label="input-box"
    type="text"
    value=""
  />
  <button
    aria-label="find-articles"
  >
    Submit
  </button>
</form>
`;
